core/js/src/span_identifier_v1.ts

```diff
@@ -0,0 +1,172 @@
+// Mirror of core/py/src/braintrust_core/span_identifier_v1.py.
```

Comment:
Just a copy of the old `span_identifier.ts` file with everything renamed to suffix with `V1`